### PR TITLE
Convert assert to verify in swarm.neo.client

### PR DESCRIPTION
**Pull request by david-eckardt-sociomantic**
Part of #200.

### CODE REVIEW

File: src/swarm/neo/client/Connection.d [hunk 1]
```diff
@@ -480,9 +480,14 @@ public final class Connection: ConnectionBase
                  */
                 return false;
 
-            default:
+            case this.status_.Connected, this.status_.Shutdown:
```
Comment (gavin-norman-sociomantic) [inline src/swarm/neo/client/Connection.d:485]: >`assert(false)` for unexpected values are converted to `verify(false)` in the following commits.

That's a nice way of handling it :+1:.

File: src/swarm/neo/client/requests/Stats.d [hunk 2]
```diff
@@ -107,7 +109,7 @@ public class Stats : IRequestStats
         ulong start_time )
     {
         auto end_time = MicrosecondsClock.now_us();
-        assert(start_time <= end_time);
```
Comment (nemanja-boric-sociomantic) [inline src/swarm/neo/client/requests/Stats.d:110]: Happy New Year (and the clock readjustment)! This could be in the separate commit.

Comment (david-eckardt-sociomantic) [reply src/swarm/neo/client/requests/Stats.d:110]: Yeah... `verify` or `enforce`, who cares, really...
(Strictly speaking, this is even a bugfix, it was asserting on input data ;)

File: src/swarm/neo/client/RequestSet.d [hunk 3]
```diff
@@ -349,18 +351,15 @@ public final class RequestSet: IRequestSet
...
 
         public void startOnNewConnection ( )
-        in
         {
-            assert(this.id);
-            assert(this.type == Type.multi_node);
+            verify(this.type == Type.multi_node);
 
-            auto rq = this.id in this.outer.active_requests;
+            assert(this.id);
```
Comment (gavin-norman-sociomantic) [inline src/swarm/neo/client/RequestSet.d]: I feel it'd be a good idea to add a small comment stating where `assert`s have already been validated with `verify` or `enforce`. Otherwise, there's always the question "was this missed?".

Comment (david-eckardt-sociomantic) [reply src/swarm/neo/client/RequestSet.d]: Actually, this is that kind of `assert` that is needed only during early debugging stages, it is much like an `invariant`. But you're right, it deserves a comment.

File: src/swarm/neo/client/requests/Stats.d (hunk 2 above)
Comment (david-eckardt-sociomantic) [reply src/swarm/neo/client/requests/Stats.d:110]: I moved it to a separate commit.

**Comment by david-eckardt-sociomantic** [comment]
Updated with comments for remaining `assert`s and an extra commit for `enforce`.

**Review by gavin-norman-sociomantic — APPROVED** [review]

**Review by nemanja-boric-sociomantic — APPROVED** [review]
